cmd/re/main.go

```diff
@@ -0,0 +1,259 @@
+	mfSDK "github.com/mainflux/mainflux/pkg/sdk/go"
+	opentracing "github.com/opentracing/opentracing-go"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+	jconfig "github.com/uber/jaeger-client-go/config"
+)
+
+const (
+	defLogLevel        = "info"
+	defHTTPPort        = "9099"
+	defKuiperURL       = "http://localhost:9081"
+	defJaegerURL       = ""
+	defServerCert      = ""
+	defServerKey       = ""
+	defSingleUserEmail = ""
+	defSingleUserToken = ""
+	defAuthURL         = "localhost:8181"
+	defAuthTimeout     = "1s"
+	defClientTLS       = "false"
+	defCACerts         = ""
+	defThingsLocation  = "http://localhost"
+	defMfBSURL         = "http://localhost:8202/things/configs"
+	defMfCertsURL      = "http://localhost:8204"
+	defTLS             = "false"
+
+	envLogLevel        = "MF_RE_LOG_LEVEL"
+	envHTTPPort        = "MF_RE_HTTP_PORT"
+	envKuiperURL       = "MF_KUIPER_URL"
+	envJaegerURL       = "MF_JAEGER_URL"
+	envServerCert      = "MF_RE_SERVER_CERT"
+	envServerKey       = "MF_RE_SERVER_KEY"
+	envSingleUserEmail = "MF_RE_SINGLE_USER_EMAIL"
+	envSingleUserToken = "MF_RE_SINGLE_USER_TOKEN"
+	envAuthURL         = "MF_AUTH_GRPC_URL"
+	envAuthTimeout     = "MF_AUTH_GRPC_TIMEOUT"
+	envClientTLS       = "MF_RE_CLIENT_TLS"
+	envCACerts         = "MF_RE_CA_CERTS"
+	envThingsLocation  = "MF_RE_THINGS_LOCATION"
+	envMfBSURL         = "MF_RE_BS_SVC_URL"
+	envMfCertsURL      = "MF_RE_CERTS_SVC_URL"
+	envTLS             = "MF_RE_ENV_CLIENTS_TLS"
+)
+
+type config struct {
+	logLevel        string
+	httpPort        string
+	kuiperURL       string
+	authHTTPPort    string
+	authGRPCPort    string
+	jaegerURL       string
+	serverCert      string
+	serverKey       string
+	singleUserEmail string
+	singleUserToken string
+	authURL         string
+	authTimeout     time.Duration
+	clientTLS       bool
+	caCerts         string
+
+	ThingsLocation string
+	MfBSURL        string
+	MfCertsURL     string
+	TLS            bool
+}
+
+func main() {
+	cfg := loadConfig()
+
+	logger, err := logger.New(os.Stdout, cfg.logLevel)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+
+	authTracer, authCloser := initJaeger("auth", cfg.jaegerURL, logger)
+	defer authCloser.Close()
+	auth, _ := createAuthClient(cfg, authTracer, logger)
+
+	tracer, reCloser := initJaeger("re", cfg.jaegerURL, logger)
+	defer reCloser.Close()
+
+	SDKCfg := mfSDK.Config{
+		BaseURL:           cfg.ThingsLocation,
+		BootstrapURL:      cfg.MfBSURL,
+		CertsURL:          cfg.MfCertsURL,
+		HTTPAdapterPrefix: "http",
+		MsgContentType:    "application/json",
+		TLSVerification:   cfg.TLS,
+	}
+	SDK := mfSDK.NewSDK(SDKCfg)
+
+	svc := newService(cfg.kuiperURL, auth, SDK, logger)
+	errs := make(chan error, 2)
+
+	go startHTTPServer(rehttpapi.MakeHandler(tracer, svc), cfg.httpPort, cfg, logger, errs)
+
+	go func() {
+		c := make(chan os.Signal)
+		signal.Notify(c, syscall.SIGINT)
+		errs <- fmt.Errorf("%s", <-c)
+	}()
+
+	err = <-errs
+	logger.Error(fmt.Sprintf("Re service terminated: %s", err))
```

Comment:
what does this mean `Re service`

Reply:
This will be eventually changes to `rules`. We are waiting on @drasko for this.

Reply:
`re` is Rules Engine. I prefer `re`, but if we want we can call it `rules`.

Reply:
I think @nmarcetic mentioned that he would prefer `rules`. What about @manuio @dusanb94 and @blokovi?

Reply:
Also, although service is called `re` in the code, you should write "Rules Engine" in the logs like this to remove ambiguity

Reply:
I vote for `rules` and I agree with @drasko about logs.